the-collab-lab/tcl-35-smart-shopping-list
# EU-DC-Mark item purchased

## Description

By ticking the checkbox of an item, the item stays checked for 24 hours.

## Acceptance Criteria

- User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
- Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates
- Also updated the user interface on first time load. Token is generated in storage, but not immediately sent to the database.
- Fixed input validation bug that previously allowed first entry of a list to be blank. This was caused by two issues: 1) input validation was performed only on lists that was NOT empty,  2) because the local storage token that identifies the list is not immediately sent to database, any input validation on an item of a list that is yet to exist in the database was also circumvented.

## Testing Steps / QA Criteria

- Create a new list.
- Add items.
- Check(mark) one/more items on the list.
- Now get out your trusty calculator and do the following: 
1. Check an item, and note the item's `lastPurchase` value is 1634547931018.
2. Dial it back a day: 1634547931018 - 86,400,000. The item should now be unchecked.
3. Add 5 minutes back: 1634547931018 - 86,400,000 + 300,000. The item should now be checked again.
